### DISCUSSION TITLE
sensor calibration and offset registers values

### DISCUSSION THREAD

**Discussion by HassanMohamed98** [jessebraham/icm42670#1] [General]
Hi,
I am working with ICM 42670-p and i want to make a calibration or want to know how to get the offset values.
thanks in advance,

**Comment by jessebraham** [comment]
Currently this driver is quite limited; right now it only supports configuring:

- Power mode
- Accel & Gyro range
- Accel & Gyro ODR

If you can elaborate on what calibration you would like to do I'm more than happy to implement the feature(s), or I'd also gladly accept a PR.

I'm not entirely sure what you're referring to by "offset values", sorry. If this is just the integer used for bit-shifting, then you just need to find the appropriate register(s) in the [data sheet](https://3cfeqx1hf82y3xcoull08ihx-wpengine.netdna-ssl.com/wp-content/uploads/2021/07/DS-000451-ICM-42670-P-v1.0.pdf).

**Reply by HassanMohamed98** [reply]
thank you,
when i read the values from the accel and gyro it has a small offset values in every direction so i think the sensor needs to be calibrated first to work properly

**Reply by jessebraham** [reply]
Okay thank you for clarifying. I will look into that when I get a chance!

**Reply by jessebraham** [reply]
I found an existing driver for this device written in C, so I will try to recreate their initialization process and see if that helps. This driver seems to be doing quite a bit more than mine does for this step.

For reference, if you're interested:  
https://github.com/zephyrproject-rtos/zephyr/tree/main/drivers/sensor/icm42670

**Reply by HassanMohamed98** [reply]
thank you, i am working in C so i think it will help me thank you again!!

**Reply by sharmatej** [reply]
how can we calibrate the sensor ?
